### PR TITLE
Fix/pagination

### LINKED ISSUE #48: How can I add pagination?
Hi there. Awesome component! Even after adding babel-regenerator-runtime, I run in exactly the problem mentioned at https://github.com/spatie/vue-table-component/issues/40
So I did this, which works fine except I don't see the pagination. My pagination is in response.data.pagination, how do I transmit it to the component ?
If I try this.table = response.data (which contains the data and pagination arrays) I get the error Invalid prop: type check failed for prop "data". Expected Array, Function, got Object.
```
<template>
       <table-component :data="table">
           <table-column show="id" label="id" data-type="numeric"></table-column>
           <table-column show="key" label="key"></table-column>
           <table-column show="lg" label="lg"></table-column>
           <table-column show="text" label="text"></table-column>
       </table-component>
</template>

<script>
    export default {
        props: {
            'url':{}
        },
        data: function() {
            return {
                'table':[]
            }
        },
        methods: {
            fetchData() {
                axios.get(this.url).then(response => {
                    console.log(response.data)
                    this.table = response.data.data
                })
            }
        },
        mounted() {
            this.fetchData()
        }
    }
</script>
```
Alternatively, if you make an example of having the async fetchData work without the regenerator bug, that might solve the issue!

### PR DESCRIPTION
**Pull request by alexankit**
Fixes #48 

pagination

Let me know if i missed something.

### CODE REVIEW

**Comment by sebastiandedeyne** [comment]
Hi, thanks for this PR!

I'm planning to merge this but gotta find some time to review it first, I'll keep you posted :)

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,6 @@
 node_modules
 npm-debug.log*
 yarn.lock
+.idea/
```
Comment (freekmurze) [inline .gitignore]: Remove these entries. They belong in a global gitignore.

File: src/components/Pagination.vue [hunk 2]
```diff
@@ -21,9 +34,53 @@
...
+                if (this.pagination.totalPages === undefined) {
+                    return [];
+                }
+
+                const pages = [];
+                let preDots = false;
+                let postDots = false;
+
+                for (let i = 1; i <= this.pagination.totalPages; i++) {
+                    if (this.pagination.totalPages <= 10) {
```
Comment (freekmurze) [inline src/components/Pagination.vue]: We're not big fans of `else` statements.

Could you refactor the loop body to it's own dedicated function so early returns can be used?

Comment (alexankit) [reply src/components/Pagination.vue]: I've created a dedicated function to generate pagination links.

Let me know if i missed something.

Comment (alexankit) [reply src/components/Pagination.vue]: I made some more improvements to the pagination. Please take a look and let me know if i missed something.

**Comment by JayBizzle** [comment]
@freekmurze you got any upcoming free time to review this and hopefully merge soon? 👍 

No problem if not, will just fork and merge this change myself for the project we need it for that is going live in the next couple of days (we have over 4,000 paginated results 😄 )

Thanks!

**Comment by JayBizzle** [comment]
We have merged this into our production app and have not encountered any issues....

...perhaps we can get this merged soon?

**Comment by sebastiandedeyne** [comment]
Sorry, lost track of this! One more thing, can this be disabled by default via a prop? So not just based on pages? Otherwide it could break existing layouts.

**Comment by alexankit** [comment]
@sebastiandedeyne I'm just modifying the page links so it should not break existing layouts.

**Comment by ankurk91** [comment]
I don't see any issue with layouts as this is already using default bootstrap css classes.

**Comment by sebastiandedeyne** [comment]
Gonna merge this tomorrow!

**Comment by JayBizzle** [comment]
> Gonna merge this tomorrow!

_/gentle nudge_ 😄

**Comment by sebastiandedeyne** [comment]
Released `1.9.0` :)

**Pull request merged**